File: files/2025-09-07/CSEC 07092025.py
import pandas as pd

path = "files/2025-09-07/Challenge 57.xlsx"
input = pd.read_excel(path, usecols="B:D", skiprows=1, nrows=6)
test  = pd.read_excel(path, usecols="F:K", skiprows=1, nrows=3)

input_long = (
    input.assign(**{'Assigned Staff': input['Assigned Staff'].str.split(', ')})
    .explode('Assigned Staff')
    .dropna(subset=['Assigned Staff'])
)
input_long['rn'] = input_long.groupby('Assigned Staff').cumcount() + 1
result = (
    input_long.pivot(index='rn', columns='Assigned Staff', values='Activity Code')
    .reindex(columns=test.columns)
    .reset_index(drop=True)
)
result.columns.name = None

print(result.equals(test)) # True

File: files/2025-09-07/CSEC 07092025.R
library(tidyverse)
library(readxl)

path = "files/2025-09-07/Challenge 57.xlsx"
input = read_excel(path, range = "B2:D8")
test  = read_excel(path, range = "F2:K5")

result = input %>%
  separate_longer_delim(`Assigned Staff`, delim = ", ") %>%
  mutate(rn = row_number(), .by = `Assigned Staff`) %>%
  select(-Activity) %>%
  na.omit() %>%
  pivot_wider(names_from = `Assigned Staff`, values_from = `Activity Code`) %>%
  select(colnames(test))

all.equal(result, test)
# [1] TRUE